src/lib.rs

```diff
@@ -22,47 +22,56 @@
 //! # }
-//! # impl embedded_hal::blocking::spi::Write<u8> for  DummySpi {
-//! #   type Error = ();
-//! #   fn write(&mut self, data: &[u8]) -> Result<(), Self::Error> { Ok(()) }
+//! # impl embedded_hal::spi::SpiDevice<u8> for DummySpi {
+//! #   fn transaction(&mut self, operations: &mut [Operation<'_, u8>]) -> Result<(), Self::Error> { Ok(()) }
+//! #   fn read(&mut self, buf: &mut [u8]) -> Result<(), Self::Error> { Ok(()) }
+//! #   fn write(&mut self, buf: &[u8]) -> Result<(), Self::Error> { Ok(()) }
+//! #   fn transfer(&mut self, read: &mut [u8], write: &[u8]) -> Result<(), Self::Error> { Ok(()) }
+//! #   fn transfer_in_place(&mut self, buf: &mut [u8]) -> Result<(), Self::Error> { Ok(()) }
 //! # }
-//! # impl embedded_hal::digital::v2::OutputPin for DummyCsPin {
-//! #   type Error = ();
-//! #   fn set_low(&mut self) -> Result<(), ()> { Ok(()) }
-//! #   fn set_high(&mut self) -> Result<(), ()> { Ok(()) }
+//! # impl embedded_hal::digital::ErrorType for DummyCsPin {
+//! #   type Error = Infallible;
+//! # }
+//! # impl embedded_hal::digital::OutputPin for DummyCsPin {
+//! #   fn set_low(&mut self) -> Result<(), Self::Error> { Ok(()) }
+//! #   fn set_high(&mut self) -> Result<(), Self::Error> { Ok(()) }
 //! # }
 //! # impl embedded_sdmmc::TimeSource for DummyTimeSource {
 //! #   fn get_timestamp(&self) -> embedded_sdmmc::Timestamp { embedded_sdmmc::Timestamp::from_fat(0, 0) }
 //! # }
-//! # impl embedded_hal::blocking::delay::DelayUs<u8> for DummyDelayer {
-//! #   fn delay_us(&mut self, us: u8) {}
+//! # impl embedded_hal::delay::DelayNs for DummyDelayer {
+//! #   fn delay_ns(&mut self, ns: u32) {}
+//! # }
+//! # impl fmt::Write for DummyUart {
+//! #     fn write_str(&mut self, s: &str) -> fmt::Result { Ok(()) }
 //! # }
-//! # impl std::fmt::Write for DummyUart { fn write_str(&mut self, s: &str) -> std::fmt::Result { Ok(()) } }
-//! # use std::fmt::Write;
 //! # use embedded_sdmmc::VolumeManager;
 //! # fn main() -> Result<(), embedded_sdmmc::Error<embedded_sdmmc::SdCardError>> {
 //! # let mut sdmmc_spi = DummySpi;
 //! # let mut sdmmc_cs = DummyCsPin;
 //! # let time_source = DummyTimeSource;
 //! # let delayer = DummyDelayer;
-//! let sdcard = embedded_sdmmc::SdCard::new(sdmmc_spi, sdmmc_cs, delayer);
-//! println!("Card size is {} bytes", sdcard.num_bytes()?);
-//! let mut volume_mgr = embedded_sdmmc::VolumeManager::new(sdcard, time_source);
-//! let mut volume0 = volume_mgr.open_volume(embedded_sdmmc::VolumeIdx(0))?;
-//! println!("Volume 0: {:?}", volume0);
-//! let mut root_dir = volume0.open_root_dir()?;
-//! let mut my_file = root_dir.open_file_in_dir("MY_FILE.TXT", embedded_sdmmc::Mode::ReadOnly)?;
-//! while !my_file.is_eof() {
-//!     let mut buffer = [0u8; 32];
-//!     let num_read = my_file.read(&mut buffer)?;
-//!     for b in &buffer[0..num_read] {
-//!         print!("{}", *b as char);
-//!     }
-//! }
+//! # let sdcard = embedded_sdmmc::SdCard::new(sdmmc_spi, sdmmc_cs, delayer);
```

Comment:
This is the example code we want to see in the `cargo doc` output, so I don't think we should hide it with `#` marks?

Reply:
I have removed the #'s :)